exporters/webhook/plugins/pelorus_handler.py

```diff
@@ -35,10 +38,100 @@
     HTTPException,
     PelorusWebhookPlugin,
     PelorusWebhookResponse,
-    Request,
 )
 
 
+def _verify_payload_signature(
+    secret: bytes, signature_secret: str, json_payload_data: Dict[str, str]
+) -> bool:
+    """
+    This function attempts to match the hash of a payload to its data,
+    with the understanding that the input JSON may be formatted slightly
+    differently, such as having different separators or newlines.
+
+    Any change to a single character in the sender's JSON input will
+    change its hash signature. Therefore, on the webhook side,
+    this function attempts to ignore separators and newline characters
+    in order to match the incoming hash using variations of those formats
+    that do not affect the payload data.
+
+    To illustrate this point, consider the following variations of
+    the same JSON input, which are actually equivalent:
+
+    {"data": "value", "data2": "value2"}
+    {"data":"value","data2":"value2"}
+    {"data":"value", "data2":"value2"}
+    {"data" :"value","data2" :"value2"}
+    {"data" :"value", "data2" :"value2"}
+    {"data": "value", "data2": "value2"}\n
+    {"data":"value","data2":"value2"}\n
+    {"data":"value", "data2":"value2"}\n
+    {"data" :"value","data2" :"value2"}\n
+    {"data" :"value", "data2" :"value2"}\n
+    { "data": "value", "data2": "value2" }
+    { "data":"value","data2":"value2" }
+    { "data":"value", "data2":"value2" }
+    { "data" :"value","data2" :"value2" }
+    { "data" :"value", "data2" :"value2" }
+    { "data": "value", "data2": "value2" }\n
+    { "data":"value","data2":"value2" }\n
+    { "data":"value", "data2":"value2" }\n
+    { "data" :"value","data2" :"value2" }\n
+    { "data" :"value", "data2" :"value2" }\n
+
+    Returns:
+        bool: True when the matching hash was found, False otherwise
+    """
+
+    # Formats:
+    #  {"data": "value", "data2": "value2"}
+    #  {"data":"value","data2":"value2"}
+    #  {"data":"value", "data2":"value2"}
+    #  {"data" :"value","data2" :"value2"}
+    #  {"data" :"value", "data2" :"value2"}
+    separator_formats = [
+        (", ", ": "),
+        (",", ":"),
+        (", ", ":"),
+        (",", " :"),
+        (", ", " :"),
+    ]
+    additional_transformation = [
+        "extra_spaces_no_newline",
+        "extra_spaces_newline",
+        "no_extra_spaces_no_newline",
+        "no_extra_spaces_newline",
+    ]
+
+    for separator in separator_formats:
+        for idx, transformation in enumerate(additional_transformation):
+            # no_extra_spaces_no_newline - default
+            payload_json_string = json.dumps(
+                json_payload_data, separators=separator, indent=None
+            )
+            payload_json_string.rstrip("\n")
```

Comment:
I think this line does not do anything, can you check it out?

I believe you would have to do `payload_json_string = payload_json_string.rstrip("\n")` to give the wanted result

Reply:
Good catch ! You are correct. Fixed.